rep-2013.rst

```diff
@@ -0,0 +1,103 @@
+REP: 2013
+Title: ROS 2 Client Library Integration
+Author: Steven! Ragnarok <steven@openrobotics.org>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 21-Sep-2022
+Post-History: 21-Sep-2022
+
+
+Abstract
+========
+
+This REP will document, retroactively, the aspects and expectations of language-specific client libraries in ROS 2 in order to serve as a template for new language communities and client libraries which can be integrated into ROS.
+In particular this REP is an attempt to elaborate on the questions raised on ROS Discourse [1]_ regarding incorporating third-party client libraries into the ROS 2 core.
+
+
+Integration and Infrastructure
+==============================
+
+rosidl pipeline
+^^^^^^^^^^^^^^^
+
+TODO introduce the idl pipeline in a way that is accurate and succinct.
+Message packages in ROS contain generated code for supported languages to directly interact with message fields.
+The rosidl pipeline handles the process of generating language specific code for working with each supported runtime and typesupport.
+Generating and distributing code for a rosidl runtime on the ROS build farm requires the rosidl runtime and typesupport packages to be available when message packages are built.
+
+package build types
+^^^^^^^^^^^^^^^^^^^
+
+TODO hoist in some of the stuff common to colcon and bloom
+
+colcon
+^^^^^^
+
+colcon is the recommended build tool for ROS 2.
+It is highly extensible and support for nearly all package build types is provided by extensions rather than built into colcon core.
+Support for most build types is also separated into base support and ros-specific support so that colcon can be used with that build type with or without ROS.
+The prime examples here are colcon-cmake [2]_ and colcon-ros-cmake [3]_ which provide general support for CMake packages and some ROS specific behavior which builds on the existing CMake support.
+
+rosdep
+^^^^^^
+
+If your client library is associated with a language-specific package manager, it can be useful to support that language specific package manager in rosdep.
+In order for a package to be released on the ROS build farm, all of its dependencies must be satisifed either in the ROS distribution directly (sometimes referred to as vendoring) or available from the platform package manager.
+However, it is still useful to have rosdep support while packages are being developed and to support community members who may be working on packages that do not need to be released on the ROS build farm.
+
+
+bloom
+^^^^^
+
```

Comment:
It would be nice to add a leading line that says what bloom is and maybe to have a link to it.